docusaurus/docs/setting-up-your-editor.md

```diff
@@ -43,10 +35,47 @@ If you're using TypeScript and Visual Studio Code, the [ESLint Visual Studio Cod
 
 Now your editor should report the linting warnings.
 
-Note that even if you edit your `.eslintrc.json` file further, these changes will **only affect the editor integration**. They won’t affect the terminal and in-browser lint output. This is because Create React App intentionally provides a minimal set of rules that find common mistakes.
+Note that even if you customise the ESLint config, these changes will **only affect the editor integration**. They won’t affect the terminal and in-browser lint output. This is because Create React App intentionally provides a minimal set of rules that find common mistakes.
 
 If you want to enforce a coding style for your project, consider using [Prettier](https://github.com/jlongster/prettier) instead of ESLint style rules.
 
+### Experimental: Extending the ESLint config
+
+We recognise that in some cases, further customisation is required. It is now possible to extend the base ESLint config by setting the `EXTEND_ESLINT` environment variable to `true`. See [advanced configuration](advanced-configuration.md) for more information on available environment variables.
+
+Note that any rules set to `"error"` will stop the project from building.
+
+There are a few things to remember:
+
+1. This currently only supports configs inside of `package.json`. This is already set for you when creating a new project.
+1. You must still extend the base config, as removing it could introduce hard-to-find issues.
+1. When working with TypeScript, you'll need to provide an `overrides` object for rules that should _only_ target TypeScript files.
+
+In the below example:
+
+- the base config has been extended by a shared ESLint config,
+- a new rule has been set that applies to all JavaScript and TypeScript files, and
+- a new rule has been set that only targets TypeScript files.
+
+```json
+{
+  "eslintConfig": {
+    "extends": ["react-app", "shared-config"],
```

Comment:
There may be an issue here for users that don't want to define their rules in `package.json`. For users that prefer to use `.eslintrc.json` or the `.js` equivalent, I believe they would have to define like so:

```json
{
  "eslintConfig": {
    "extends": ["react-app", "./eslintrc"]
  }
}
```

However, now they would probably not use `extends: ["react-app"]` in their `.eslintrc.json` since it's already defined here, which i believe would exclude the `react-app` rules from their editor integration? In my experience I believe the VS Code ESLint extension will use the `.eslintrc.json` file if it exists. If this is the case that might be fairly clunky, and forcing the rules to be in `package.json` doesn't seem ideal (I personally would rather not bloat the `package.json` more than it already is in my projects).

Thoughts?

Reply:
>Thoughts?

Specifically on `package.json` - I don't like putting linting rules in there specifically.  I tend to comment why rules are the way they are (or aren't), and package.json doesn't allow such practices.

Reply:
Yes, I agree @ianschmitz and @Asday - my thought was that as this is experimental, we could require it to be in `package.json`, and when we enable by default, we can expand to all configs.

What do you think? We could just get it in now, I just don't want to confuse things too much.

If we support all formats, we should also alert people if they have multiple configs.